src/audio_unit/macos_helpers.rs

```diff
@@ -50,16 +49,16 @@ pub fn get_default_device_id(input: bool) -> Option<AudioDeviceID> {
         mElement: kAudioObjectPropertyElementMaster,
     };
 
-    let audio_device_id: AudioDeviceID = 0;
-    let data_size = mem::size_of::<AudioDeviceID>();
+    let mut audio_device_id: AudioDeviceID = 0;
+    let data_size = mem::size_of::<AudioDeviceID>() as u32;
     let status = unsafe {
         AudioObjectGetPropertyData(
-            kAudioObjectSystemObject,
-            &property_address as *const _,
+            kAudioObjectSystemObject as AudioObjectID,
+            NonNull::from(&property_address),
             0,
             null(),
-            &data_size as *const _ as *mut _,
-            &audio_device_id as *const _ as *mut _,
+            NonNull::from(&data_size),
+            NonNull::from(&mut audio_device_id).cast(),
         )
     };
```

Comment:
I made sure to use `&mut` on the out pointer `audio_device_id` here (and in similar cases below), this was UB before.